tmt/config/models/hardware.py

```diff
@@ -0,0 +1,13 @@
+from typing import Optional
+
+from tmt.container import MetadataContainer
+
+
+class Translate(MetadataContainer):
+    name: str
+    child_name: Optional[str]
+    element: str
```

Comment:
Check out https://github.com/teemtee/tmt/pull/2967#discussion_r1771508347, the second example: we could replace `name` and `child_name` with just `name`, and then use `constraint.printable_name` to compare against. That would compare names like `compatible.distro` or `cpu.processors`, which would be readable.

Also, I'd change `element` to something like `element_template` or just `template`.